### ISSUE TITLE
Ansible via Docker on Windows

### ISSUE THREAD

**Issue by rufflabs** [Orange-Cyberdefense/GOAD#101]
I am using Win11 with VMware Workstation. I have all the VM's up via Vagrant and decided to try using the Docker container for Ansible. I built the container in my Win11 host using Docker Desktop and the container was built and brought up. However, it can't seem to connect to the VM's. 

I suspect this may be because the VM's are in a VMware network while the Ansible container is on Docker which I believe is running within WSL. 

Anyone experienced this or know what I might be able to do? Otherwise I'm going to just run ansible from another linux VM within the same vmnet. 

```
PS C:\Code\GOAD> docker run -ti --rm --network host -h goadansible -v c:\code\goad:/goad -w /goad/ansible goadansible ansible-playbook -i ../ad/sevenkingdoms.local/inventory main.yml
[WARNING]: Ansible is being run in a world writable directory (/goad/ansible), ignoring it as an ansible.cfg source. For more information see
https://docs.ansible.com/ansible/devel/reference_appendices/config.html#cfg-in-world-writable-dir

PLAY [Read data files] ************************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************************
fatal: [dc01]: UNREACHABLE! => {"changed": false, "msg": "ssl: HTTPSConnectionPool(host='192.168.56.10', port=5986): Max retries exceeded with url: /wsman (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc2898719c0>: Failed to establish a new connection: [Errno 111] Connection refused'))", "unreachable": true}
fatal: [dc02]: UNREACHABLE! => {"changed": false, "msg": "ssl: HTTPSConnectionPool(host='192.168.56.11', port=5986): Max retries exceeded with url: /wsman (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc289871d50>: Failed to establish a new connection: [Errno 111] Connection refused'))", "unreachable": true}
fatal: [srv02]: UNREACHABLE! => {"changed": false, "msg": "ssl: HTTPSConnectionPool(host='192.168.56.22', port=5986): Max retries exceeded with url: /wsman (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc289871d80>: Failed to establish a new connection: [Errno 111] Connection refused'))", "unreachable": true}
fatal: [dc03]: UNREACHABLE! => {"changed": false, "msg": "ssl: HTTPSConnectionPool(host='192.168.56.12', port=5986): Max retries exceeded with url: /wsman (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc289871d80>: Failed to establish a new connection: [Errno 111] Connection refused'))", "unreachable": true}
fatal: [srv03]: UNREACHABLE! => {"changed": false, "msg": "ssl: HTTPSConnectionPool(host='192.168.56.23', port=5986): Max retries exceeded with url: /wsman (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fc289871d80>: Failed to establish a new connection: [Errno 111] Connection refused'))", "unreachable": true}

PLAY RECAP ************************************************************************************************************************************************
dc01                       : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
dc02                       : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
dc03                       : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
srv02                      : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
srv03                      : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
```

**Comment by an4kein** [comment]
This solution does not work for me!
But this solution does, you need to install a certificate on the server and allow port 5986 for winrmHTTPS.

1 - Configure using the administrator account's IP Address.
2 - Install Certificate.
3 - Allow port 5986 on the firewall.

Before:
![image](https://github.com/Orange-Cyberdefense/GOAD/assets/37910997/927832be-909d-4941-a83a-cb6c52837faf)

After:
![image](https://github.com/Orange-Cyberdefense/GOAD/assets/37910997/b20d6ba9-682f-4d28-9013-6b7c466f5258)

Creating a Self-Signed Certificate
`New-SelfSignedCertificate -Subject 'CN=ServerB.domain.com' -TextExtension '2.5.29.37={text}1.3.6.1.5.5.7.3.1'`

Configuring the WinRM Listener
`winrm create winrm/config/Listener?Address=*+Transport=HTTPS '@{Hostname="ServerB.domain.com"; CertificateThumbprint="<cert thumbprint here>"}'`

Opening the WinRM SSL Firewall Port
`$FirewallParam = @{
    DisplayName = 'Windows Remote Management (HTTPS-In)'
    Direction = 'Inbound'
    LocalPort = 5986
    Protocol = 'TCP'
    Action = 'Allow'
    Program = 'System'
}
New-NetFirewallRule @FirewallParam`

REF: https://adamtheautomator.com/winrm-ssl/

**Comment by saladandonionrings** [comment]
> This solution does not work for me! But this solution does, you need to install a certificate on the server and allow port 5986 for winrmHTTPS.
> 
> 1 - Configure using the administrator account's IP Address. 2 - Install Certificate. 3 - Allow port 5986 on the firewall.
> 
> Before: ![image](https://user-images.githubusercontent.com/37910997/246983390-927832be-909d-4941-a83a-cb6c52837faf.png)
> 
> After: ![image](https://user-images.githubusercontent.com/37910997/246983428-b20d6ba9-682f-4d28-9013-6b7c466f5258.png)
> 
> Creating a Self-Signed Certificate `New-SelfSignedCertificate -Subject 'CN=ServerB.domain.com' -TextExtension '2.5.29.37={text}1.3.6.1.5.5.7.3.1'`
> 
> Configuring the WinRM Listener `winrm create winrm/config/Listener?Address=*+Transport=HTTPS '@{Hostname="ServerB.domain.com"; CertificateThumbprint="<cert thumbprint here>"}'`
> 
> Opening the WinRM SSL Firewall Port `$FirewallParam = @{ DisplayName = 'Windows Remote Management (HTTPS-In)' Direction = 'Inbound' LocalPort = 5986 Protocol = 'TCP' Action = 'Allow' Program = 'System' } New-NetFirewallRule @FirewallParam`
> 
> REF: https://adamtheautomator.com/winrm-ssl/ My config: https://pastebin.com/raw/kgm4S5f1

Thank you very much for this, it works perfectly and it saved me many hours ! :)

**Comment by Gemei** [comment]
For anyone else having this issue but only for one of the servers. In my case it was [srv03]. It turns out that the host-only interface IP was set incorrectly for some reason. I logged into the server using Vagrant user (Password is vagrant), and changed the IP to static, in my case it was 192.168.56.23

**Comment by Mayfly277** [comment]
duplicate of https://github.com/Orange-Cyberdefense/GOAD/issues/98
